src/app/_components/TextLink.tsx

```diff
@@ -3,40 +3,33 @@
 import { ArrowUpRight } from '@phosphor-icons/react'
 import { clsx } from 'clsx'
 
-import { isExternalLink } from '@/utils/linkUtils'
+import { isInternalLink } from '@/utils/linkUtils'
 
-import { CustomLink } from '@/components/CustomLink'
+import { type CustomLinkProps, CustomLink } from '@/components/CustomLink'
 import { Icon } from '@/components/Icon'
 
-type TextLinkProps = {
-  href: string
-  children: React.ReactNode
-} & React.ComponentProps<'a'>
-
 export const linkBaseStyles =
   'text-brand-300 focus:brand-outline hover:underline'
 
 export function TextLink({
```

Comment:
I find it a bit confusing that we have `TextLink` and `CustomLink` 😕 they seem to be doing similar things?

Reply:
They are not - `CustomLink` is just a shallow wrapper. `TextLink` includes styles and the arrow out icon.

Reply:
I feel like having one component would make maintenance easier...But I haven't tried consolidating, not sure what effect would that have on the code base.

Reply:
Have you checked the use cases for `CustomLink`? I've expanded my initial comment. Maybe that helps explain it.

Reply:
At some point, we might also add `OverlayLink` etc. Which would use `CustomLink` internally.

Reply:
Thanks for clarifying! It order wasn't clear to me at first

Reply:
Thank you for keeping asking. It's important that it's clear.